contracts/external/cw-admin-factory/src/contract.rs

```diff
@@ -39,17 +45,25 @@ pub fn execute(
             instantiate_msg: msg,
             code_id,
             label,
-        } => instantiate_contract(env, info, msg, code_id, label),
+        } => instantiate_contract(deps, env, info, msg, code_id, label),
     }
 }
 
 pub fn instantiate_contract(
+    deps: DepsMut,
     env: Env,
     info: MessageInfo,
     instantiate_msg: Binary,
     code_id: u64,
     label: String,
 ) -> Result<Response, ContractError> {
+    // If admin set, require the sender to be the admin.
+    if let Some(admin) = ADMIN.load(deps.storage)? {
+        if admin != info.sender {
+            return Err(ContractError::Unauthorized {});
+        }
+    }
+
     // Instantiate the specified contract with factory as the admin.
     let instantiate = WasmMsg::Instantiate {
         admin: Some(env.contract.address.to_string()),
```

Comment:
if admin is set, could it be desirable to use that address instead of the factory as the wasm instantiate msg admin?

Reply:
in the reply, this contract changes the admin (using the fact that it is the current admin) to the DAO's address. that's the point of the contract :)

Reply:
forgot to get back on this, gotcha. then all makes sense!